### PR TITLE
[PWN-4075] Feature/pwn 4075 app coordinator

### PR DESCRIPTION
**Pull request by bigearsenal**
## Link jira to issue
https://p2pvalidator.atlassian.net/browse/PWN-4075

## Description of the changes
- `Root` scene has been replaced by `AppCoordinator`, so there would be no extra `Root.ViewController` deep down.
- `AppCoordinator` will be responsible for navigating between main scenes

### CODE REVIEW

File: p2p_wallet/AppCoordinator/AppCoordinator+AppEventHandlerDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        navigateToOnboardingDone()
+    }
+
+    func userDidChangeAPIEndpoint(to _: APIEndPoint) {
+        showAuthenticationOnMainOnAppear = false
+        Task {
+            await reload()
+        }
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.3) { [weak self] in
```
Comment (lisemyon) [inline p2p_wallet/AppCoordinator/AppCoordinator+AppEventHandlerDelegate.swift:46]: why is the timing 0.3 ? is it supposed to happen after `reload()` finished ?

File: p2p_wallet/AppCoordinator/AppCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            window.rootViewController = nil
+            // add lockview
+            lockView.removeFromSuperview()
+            window.addSubview(lockView)
+            lockView.autoPinEdgesToSuperviewEdges()
+
+            // show loading
+            indicator.removeFromSuperview()
+            window.addSubview(indicator)
+            indicator.autoPinEdgesToSuperviewEdges()
```
Comment (lisemyon) [inline p2p_wallet/AppCoordinator/AppCoordinator.swift:56]: will it be better to extract this piece of `MainActor.run` code to a separate function? to simplify this method

Comment (bigearsenal) [reply p2p_wallet/AppCoordinator/AppCoordinator.swift:56]: yes of course. I'll do it

File: p2p_wallet/AppCoordinator/AppCoordinator+AppEventHandlerDelegate.swift (hunk 1 above)
Comment (bigearsenal) [reply p2p_wallet/AppCoordinator/AppCoordinator+AppEventHandlerDelegate.swift:46]: Yes. Maybe the migration code from old Root to new Coordinator cause this line. I'll fix it.
Thank you!

**Review by lisemyon — DISMISSED** [review]

**Review by sidorov-panda — DISMISSED** [review]

**Review by TrGiLong — DISMISSED** [review]

File: p2p_wallet/AppCoordinator/AppCoordinator.swift [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    private var appEventHandler: AppEventHandlerType = Resolver.resolve()
+    private let storage: AccountStorageType & PincodeStorageType & NameStorageType = Resolver.resolve()
+    let analyticsManager: AnalyticsManager = Resolver.resolve()
+    let notificationsService: NotificationService = Resolver.resolve()
+
+    // MARK: - Properties
+
+    let window: UIWindow
+    let lockView = LockView()
```
Comment (qwerkly) [inline p2p_wallet/AppCoordinator/AppCoordinator.swift:25]: Do you need to contain this in memory? I think you need to make LockView as optional and when you remove it, set it to nil also.

File: p2p_wallet/AppDelegate.swift [hunk 4]
```diff
@@ -64,16 +65,18 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
         }
 
-        notificationService.wasAppLaunchedFromPush(launchOptions: launchOptions)
-
-        // set rootVC
-        let vm = Root.ViewModel()
-        let vc = Root.ViewController(viewModel: vm)
-        window?.rootViewController = vc
+        // add placeholder
+        window?.rootViewController = BaseVC()
```
Comment (qwerkly) [inline p2p_wallet/AppDelegate.swift:69]: I think it's better to move window setting logic to coordinator and this code is going to look as this
`
appCoordinator.start()
window = appCoordinator.mainWindow
`

**Review by qwerkly — DISMISSED** [review]

**Pull request merged**